articles/112_generated_interfaces_cpp.md

```diff
@@ -28,16 +30,25 @@ This article specifies the generated C++ code for ROS interface types defined in
 ## Namespacing
 
 All code of a ROS package should be defined in a namespace named after the package.
-To separate the generated code from other code within the package it is defined in a sub namespace:
+To separate the generated code from other code within the package it is defined in a sub-namespace:
+
 * namespace for ROS messages: `package_name::msg`.
 * namespace for ROS services: `package_name::srv`.
 
+<div class="alert alert-warning" markdown="1">
+  **Note:** Those namespaces are not consistent with ROS 1 usage.
+</div>
+
```

Comment:
> Since the typename of ROS 1 message and ROS 2 messages is the same with your proposed change how shall these two types then be used in C++ code?

The intent is for them to be used the same way. When compiled for ROS 1, they will resolve to boost shared pointers. For ROS 2 they will use C++11 shared pointers. Code that explicitly uses the Boost name rather than the typedef will need to be modified, but will work with either version.

Or, is this another question about that static ROS 1 bridge prototype?